docs/src/user-guide/tutorials/auction.md

```diff
@@ -0,0 +1,349 @@
+Your method can look as follows:
+
+```rust
+pub fn action_end(mut self, mut ctx: Context) {
+    if ctx.block_timestamp() <= self.auction_end_time {
+        revert AuctionNotYetEnded()
+    }
+    if self.ended {
+        revert AuctionEndAlreadyCalled()
+    }
+    self.ended = true
+    ctx.emit(AuctionEnded(winner: self.highest_bidder, amount: self.highest_bid))
+
+    ctx.send_value(to: self.beneficiary, wei: self.highest_bid)
+}
+```
+
+Congratulations! You just wrote an open auction contract in Fe!
+
+
+## View functions
+
+To help test the contract without having to decode transaction logs, you can add some simple functions to the contract that simply report the current values for some key state variables (specifically, `highest_bidder`, `highest_bid` and `ended`). This will allow a user to use `eth_call` to query these values in the contract. `eth_call` is used for functions that do not update the state of the blockchain and costs no gas because the queries can be performed on local data.
+
+You can add the following functions to the contract:
+
+```rust
+pub fn check_highest_bidder(mut self, ctx: Context) -> address {
+    return self.highest_bidder;
+}
+
+pub fn check_highest_bid(mut self, ctx: Context) -> u256 {
+    return self.highest_bid;
+}
+
+pub fn check_ended(mut self, ctx: Context) -> bool {
+    return self.ended;
+}
+```
+
+## Build and deploy the contract
+
+Your contract is now ready to use! Compile it using
+
+```sh
+fe build auction.fe
+```
+
+You will find the contract ABI and bytecode in the newly created `outputs` directory.
+
+Start a local blockchain to deploy your contract to:
+
+```sh
+anvil
+```
+
+There are constructor arguments (`bidding_time: u256`, `beneficiary_addr: address`) that have to be added to the contract bytecode so that the contract is instantiated with your desired values. To add constructor arguments you can encode them into bytecode and append them to the contract bytecode.
+
+First, hex encode the value you want to pass to `bidding_time`. In this case, we will use a value of 10:
+
+```sh
+cast --to_hex(10)
+
+>> 0xa // this is 10 in hex
+```
+
+Ethereum addresses are already hex, so there is no further encoding required. The following command will take the constructor function and the hex-encoded arguments and concatenate them into a contiguous hex string, and save it to `constructor_args.txt`. 
+
+```sh
+cast abi-encode "__init__(uint256,address)" "0xa" "0xa0Ee7A142d267C1f36714E4a8F75612F20a79720" > constructor_args.txt
+```
+
+Now append these constructor argument bytes to the bytecode generated by `fe build` (not that the `cut -c3` command removes the leading `0x` from the bytes being appended)
+
+```sh
+cat constructor_args.txt | cut -c3- >> output/Auction/Auction.bin
```

Comment:
I wonder how we can make this part more ergonomic. Vyper should have the same problem with foundry and they also have a foundry-vyper helper which our foundry-fe helper is based on. But I see that mostly used for testing. I'm not sure yet if/how one could use that for simple deployments. I will look into that.

Reply:
I *think* if you use `cast send` with the `--create` flag then you'll be able pass the raw bytecode and will also be able to append constructor args and have foundry do the heavy lifting.

Reply:
Ok, good to know. I could also take a look

Reply:
I didn't get it to work exactly the way I expected it, but this worked for me:

```
cast send --from <whatever-anvil-says> --create $(cat output/Auction/Auction.bin) $(cast abi-encode "__init_(uint256, address)", 0xa 0xa0Ee7A142d267C1f36714E4a8F75612F20a79720)

```